packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3864,27 +3864,44 @@ export class ContainerRuntime
 	public async getPendingLocalState(props?: {
 		notifyImminentClosure: boolean;
 	}): Promise<unknown> {
-		this.verifyNotClosed();
-		const waitBlobsToAttach = props?.notifyImminentClosure;
-		if (this._orderSequentiallyCalls !== 0) {
-			throw new UsageError("can't get state during orderSequentially");
-		}
-		const pendingAttachmentBlobs = await this.blobManager.getPendingBlobs(waitBlobsToAttach);
-
-		if (!pendingAttachmentBlobs && !this.hasPendingMessages()) {
-			return; // no pending state to save
-		}
+		return PerformanceEvent.timedExecAsync(
+			this.mc.logger,
+			{
+				eventName: "getPendingLocalState",
+				notifyImminentClosure: props?.notifyImminentClosure,
+			},
+			async () => {
```

Comment:
this function will be passed an event object that you can add properties to

Reply:
you will probably use .end method, and add the final properties at the end of the method